### PR TITLE
[IMP] partner_credit_limit and payment_term_type: Migrated to v9

### PR DESCRIPTION
**Pull request by egrivero**
MIgrated to v9 the module partner_credit_limit and payment_term_type

### CODE REVIEW

**Comment by egrivero** [comment]
Hi @moylop260 

What is the error travis showing me here? I do not understand

**Comment by luistorresm** [comment]
@egrivero 

You need add this dependance cython [here](https://github.com/Vauxoo/addons-vauxoo/blob/9.0/requirements.txt)

Regards

**Comment by zaoral** [comment]
> @egrivero
> 
> You need add this dependance cython here
> 
> Regards

@luistorresm @egrivero 
FYI I already made the poc and I added the dependency but still not work. You can see this in #1074

**Comment by zaoral** [comment]
NOTE: The change I made was in 8.0 and do not fix the error. And not sure but for 9.0 I think that the result probably could be the same

**Comment by luistorresm** [comment]
Thanks @zaoral 

I was added the dependency [here](https://github.com/Vauxoo/imeqmo/pull/5/files#diff-b4ef698db8ca845e5845c4618278f29aR2) and the travis was generated.

But I do not know because here not work.

Regards

**Comment by egrivero** [comment]
Thanks @zaoral @luistorresm 
I will add it here just to prove that generates results

**Comment by moylop260** [comment]
Somebody could help me to apply the following patch https://github.com/OCA/maintainer-quality-tools/pull/355?

**Comment by egrivero** [comment]
Corrected the error it add the dependence to cython

**Comment by egrivero** [comment]
Can someone check if any changes you make in this PR are well so it can be merge?

**Comment by hbto** [comment]
@moylop260 

On the way

> Somebody could help me to apply the following patch OCA/maintainer-quality-tools#355?

**Comment by hbto** [comment]
Problem with Pandas / Cython fixed at Vauxoo/maintainer-quality-tools@a9759c9

Your travis instances shall begin to be build flawlessly.

Best Regards.

**Comment by egrivero** [comment]
Hi @moylop260, can you help me to merge this PR?

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 pandas >= 0.16.2
 cssutils >= 1.0.1
+cython >= 0.24.1
```
Comment (moylop260) [inline requirements.txt]: Remove cython

**Review by moylop260 — CHANGES_REQUESTED** [review]

**Comment by moylop260** [comment]
Could you request approvals before of merging please?

**Comment by egrivero** [comment]
Eliminate the requirement line of file, but it gives me an error travis.

I will ask for approval so that it can be merge

**Comment by egrivero** [comment]
@moylop260 

Functional review of the approved changes here Vauxoo/imeqmo#18. Can you please help me with the merge of this PR?

File: partner_credit_limit/model/partner.py [hunk 2]
```diff
@@ -59,8 +52,9 @@ def _get_overdue_credit(self):
             moveline_obj = self.env['account.move.line']
             movelines = moveline_obj.search(
                 [('partner_id', '=', partner.id),
-                 ('account_id.type', '=', 'receivable'),
-                 ('state', '!=', 'draft'), ('reconcile_id', '=', False)])
+                 ('account_id.internal_type', '=', 'receivable'),
+                 ('move_id.state', '!=', 'draft'),
```
Comment (moylop260) [inline partner_credit_limit/model/partner.py:56]: `state` field is not present?

File: partner_credit_limit/model/sale.py [hunk 3]
```diff
@@ -30,5 +23,10 @@ def check_limit(self):
...
                         'has late payments or has exceeded the credit limit.'
                         '\nPlease cover the late payment or check credit limit'
-                        '\nCreadit Limit : %s') % (so.partner_id.credit_limit)
-                raise exceptions.Warning(('Warning!'), msg)
+                        '\nCredit Limit : %s') % (so.partner_id.credit_limit)
+                raise exceptions.Warning(msg)
+
+    @api.multi
+    def action_confirm(self):
+        self.check_limit()
```
Comment (moylop260) [inline partner_credit_limit/model/sale.py:31]: Why this change is required for the migration of the module?

File: partner_credit_limit/tests/common.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class TestCommon(TransactionCase):
+
+    def setUp(self):
+        super(TestCommon, self).setUp()
+        self.account_invoice = self.env['account.invoice']
+        self.account_invoice_line = self.env['account.invoice.line']
+
+    def _load(self, module, *args):
```
Comment (moylop260) [inline partner_credit_limit/tests/common.py:17]: Is this code copied from:___?

**Review by moylop260 — APPROVED** [review]
LGTM just small questions to know the "Why"

**Comment by moylop260** [comment]
Create the pr dummy and request review to leader of the project

File: partner_credit_limit/model/partner.py (hunk 2 above)
Comment (egrivero) [reply partner_credit_limit/model/partner.py:56]: In version 9, I did not get the state field in the model account.move.line

File: partner_credit_limit/model/sale.py (hunk 3 above)
Comment (egrivero) [inline partner_credit_limit/model/sale.py]: Before the call to that function was made by the workflow, but in version 9 there is no sales workflow, so pass the validation call to the function that is responsible for confirming the sales order

Comment (egrivero) [reply partner_credit_limit/model/sale.py:31]: Before the call to that function was made by the workflow, but in version 9 there is no sales workflow, so pass the validation call to the function that is responsible for confirming the sales order

**Comment by egrivero** [comment]
The PR dummy is this Vauxoo/imeqmo#18, it is functionally approved, and I wanted it to be technically reviewed here.

File: partner_credit_limit/tests/common.py (hunk 4 above)
Comment (egrivero) [reply partner_credit_limit/tests/common.py:17]: This code copied it from [here](https://github.com/odoo/odoo/blob/9.0/addons/account_asset/tests/test_account_asset.py#L10), I needed accounting data demo that in version 9 is no already loaded, but there are the files with the information.

**Comment by moylop260** [comment]
Thanks for clarify

**Comment by egrivero** [comment]
Thanks @moylop260

**Pull request merged**